src/auto/injector.js

```diff
@@ -62,17 +62,23 @@
  * Implicit module which gets automatically added to each {@link auto.$injector $injector}.
  */
 
+var ARROW_ARG = /^([^\(]+)=>/;
```

Comment:
This RegExp would incorrectly match `arg1 => { var arrow = '=>'; }` (it would match greedily until the last `=>` before the first `(`).
The `+` should be made non-greedy: `/^([^(]+?)=>/`